src/js/angular/getImgService.test.js

```diff
@@ -0,0 +1,134 @@
+describe('getImagesService', function(){
+    'use strict';
+
+    var $http;
+    var ImgService;
+
+    beforeEach(module('flickrPOC'));
+
+    beforeEach(inject(function(_$http_, _images_){
+        $http = _$http_;
+        ImgService = _images_;
+    }));
+
+    describe('loadPics', function(){
+        describe('stub service call', function(){
+            it('should exist', function(){
+                expect(ImgService.loadPics).to.exist();
+            });
+            it('should use $http to retrieve data', function(){
+                var spy = sinon.spy($http, 'get');
+                var method = "flickr.photos.search";
+                var api_key = '74e47a159e15cbcb6139ba9c9df64c13';
+                var user_id = "138698049@N03";
+                var mockUrl = 'https://api.flickr.com/services/rest?method=' + method + '&api_key='+ api_key + '&user_id=' + user_id + '&format=json&nojsoncallback=1';
+
+                ImgService.loadPics();
+                expect(spy).to.have.been.calledWith(mockUrl);
+            });
+
+            it('should return a promise', function(){
+               expect(ImgService.loadPics().then).to.exist();
+            });
+        });
+    });
+    describe('createImagesArray', function(){
+        var sampleJSON = {contentType: "text/json",
+            status: "success",
+
+            photos: {
+                photo: [
+                    {
+                        farm: 6,
+                        id: "23780914905",
+                        isfamily: 0,
+                        isfriend: 0,
+                        ispublic: 1,
+                        owner: "138698049@N03",
+                        secret: "0eb2a77a55",
+                        server: "5764",
+                        title: ""
+                    },
+                    {
+                        farm: 1,
+                        id: "23698470321",
+                        isfamily: 0,
+                        isfriend: 0,
+                        ispublic: 1,
+                        owner: "138698049@N03",
+                        secret: "1d4dc255d0",
+                        server: "693",
+                        title: ""
+                    }
+                ]
+            }
+
+
+    };
+
+        describe('should check the data taken by createdDomElm', function(){
+            var getImages = function(){
+                return ImgService.createImagesArray(sampleJSON);
+            };
```

Comment:
I like this approach, however, I think it can be simplified a bit. 

By using a `beforeEach`, we can set a value that we can reference in each test.

```javascript
describe('should check the data taken by createdDomElm', function(){
  var imagesArray;

  beforeEach(function(){
    imagesArray = ImgService.createImagesArray(sampleJSON);
  });

  // now your tests don't have to call the custom function
  it('should return an array', function(){
    expect(imagesArray).to.be.true();
  });

  // and so on

});
```